osquery/events/tests/eventsubscriberplugin.cpp

```diff
@@ -349,9 +360,13 @@ TEST_F(EventSubscriberPluginTests, generateRowsWithExpiry) {
   subscriber.setTime(20);
   subscriber.generateRows(callback, true, 0, 0);
   EXPECT_EQ(10U, callback_count);
-  // Events are expired after being queried.
+
+  // Events are expired after being queried (except for last valid event --
+  // optimize should take care of that event not being returned again, but that
+  // is tested separately below.)
```

Comment:
I don't really understand how optimize prevents it from being returned. Or how this tests it.

Reply:
Optimization keeps track of the highest eventid and then when the generate function is called any event with a lower ID is skipped when returning rows. The test for optimization verifies that behavior.